### PR TITLE
Notification Hub SDK change

### PR DESCRIPTION
**Pull request by amolr**
This includes the recently added SKU change for namespace and also the patch operation supported for NH namespaces

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@amolr__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
@amolr seems tests are failing for notificationhub.

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/NotificationHubs/Microsoft.Azure.Management.NotificationHubs/Generated/Models/CheckAvailabilityParameters.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace Microsoft.Azure.Management.NotificationHubs.Models
     /// Parameters supplied to the Check Name Availability for Namespace and
     /// NotificationHubs.
     /// </summary>
-    public partial class CheckAvailabilityParameters : Resource
```
Comment (markcowl) [inline src/ResourceManagement/NotificationHubs/Microsoft.Azure.Management.NotificationHubs/Generated/Models/CheckAvailabilityParameters.cs]: Why does this no longer extend resource?

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/project.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "summary": "NotificationHubs.Tests Tests.",
     "tags": [ "" ],
     "projectUrl": "https://github.com/Azure/azure-sdk-for-net",
-    "licenseUrl": "https://raw.githubusercontent.com/Microsoft/dotnet/master/LICENSE",
+    "licenseUrl": "https://raw.githubusercontent.com/Microsoft/dotnet/master/LICENSE"
   },
   "buildOptions": {
```
Comment (markcowl) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/project.json]: Note that this contains breaking changes, you will need to update your version accordingly (if the library is not -preview or -prerelease)

**Review by markcowl — CHANGES_REQUESTED** [review]
A couple of comments

**Review by markcowl — APPROVED** [review]

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/project.json [hunk 3]
```diff
@@ -26,15 +26,15 @@
   },
   "dependencies": {
     "dotnet-test-xunit": "2.2.0-preview2-build1029",
-    "Microsoft.Azure.Management.NotificationHubs": "2.0.1-preview",
+    "Microsoft.Azure.Management.NotificationHubs": "2.1.0-preview",
     "Microsoft.Azure.ResourceManager": "1.0.0-preview",
-    "Microsoft.Azure.Test.HttpRecorder": "[1.6.7-preview,2.0.1)",
+    "Microsoft.Azure.Test.HttpRecorder": "[1.6.7-preview,2.1.0)",
```
Comment (markcowl) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/project.json]: WHy did you change this?

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/project.json [hunk 4]
```diff
@@ -26,15 +26,15 @@
...
     "Microsoft.NETCore.App": {
       "type": "platform",
       "version": "1.0.0"
     },
     "Microsoft.Rest.ClientRuntime.Azure": "[3.3.1,4.0.0)",
-    "Microsoft.Rest.ClientRuntime.Azure.TestFramework": "[1.3.5-preview,2.0.1)",
+    "Microsoft.Rest.ClientRuntime.Azure.TestFramework": "[1.3.5-preview,2.1.0)",
```
Comment (markcowl) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/project.json]: should not be changed

**Pull request merged**